sdk/eventgrid/Microsoft.Azure.CloudNative.CloudEvents.EventGrid/tests/CloudEventTests.cs

```diff
@@ -0,0 +1,137 @@
+                        new Uri("http://localHost"),
+                        Guid.NewGuid().ToString(),
+                        DateTime.Now);
+
+                if (inclTraceparent && inclTracestate && i % 2 == 0)
+                {
+                    cloudEvent.GetAttributes().Add("traceparent", "traceparentValue");
+                }
+                if (inclTracestate && i % 2 == 0)
+                {
+                    cloudEvent.GetAttributes().Add("tracestate", "param:value");
+                }
+                eventsList.Add(cloudEvent);
+            }
+            await client.SendCloudEventsAsync(eventsList);
+
+            activity.Stop();
+            List<CloudEvent> cloudEvents = DeserializeRequest(mockTransport.SingleRequest);
+            IEnumerator<CloudEvent> cloudEnum = eventsList.GetEnumerator();
+            foreach (CloudEvent cloudEvent in cloudEvents)
+            {
+                cloudEnum.MoveNext();
+                IDictionary<string, object> cloudEventAttr = cloudEnum.Current.GetAttributes();
+                if (cloudEventAttr.ContainsKey(TraceParentHeaderName) &&
+                    cloudEventAttr.ContainsKey(TraceStateHeaderName))
+                {
+                    Assert.AreEqual(
+                        cloudEventAttr[TraceParentHeaderName],
+                        cloudEvent.GetAttributes()[TraceParentHeaderName]);
+
+                    Assert.AreEqual(
+                        cloudEventAttr[TraceStateHeaderName],
+                        cloudEvent.GetAttributes()[TraceStateHeaderName]);
+                }
+                else if (cloudEventAttr.ContainsKey(TraceParentHeaderName))
+                {
+                    Assert.AreEqual(
+                        cloudEventAttr[TraceParentHeaderName],
+                        cloudEvent.GetAttributes()[TraceParentHeaderName]);
+                }
+                else if (cloudEventAttr.ContainsKey(TraceStateHeaderName))
+                {
+                    Assert.AreEqual(
+                       cloudEventAttr[TraceStateHeaderName],
+                       cloudEvent.GetAttributes()[TraceStateHeaderName]);
+                }
+                else
+                {
+                    Assert.AreEqual(
+                       activity.Id,
+                       cloudEvent.GetAttributes()[TraceParentHeaderName]);
+                }
+            }
+        }
+
+        private static List<CloudEvent> DeserializeRequest(Request request)
+        {
+            var content = request.Content;
+            var stream = new MemoryStream();
+            content.WriteTo(stream, CancellationToken.None);
+            stream.Position = 0;
+            JsonDocument requestDocument = JsonDocument.Parse(stream);
+            var cloudEvents = new List<CloudEvent>();
+            // Parse JsonElement into separate events, deserialize event envelope properties
+            if (requestDocument.RootElement.ValueKind == JsonValueKind.Object)
```

Comment:
Can you help me understand the case where we get a RootElement.ValueKind of object here?  I thought that the top level object will always be an array (even if a single event is provided)?

Reply:
This was carried forward from the Track 1 library deserialization handling. It is definitely not needed here as the input is the array that was constructed when publishing. Will remove.